src/Test/Hspec/Yesod.hs

```diff
@@ -601,18 +601,44 @@ statusIs :: HasCallStack => Int -> YesodExample site ()
 statusIs number = do
   latestRequest <- requireLatestRequest
   withResponse $ \(SResponse status headers body) -> do
-    let mContentType = lookup hContentType headers
-        isUTF8ContentType = maybe False YT.Internal.contentTypeHeaderIsUtf8 mContentType
-
-    liftIO $ flip HUnit.assertBool (H.statusCode status == number) $ concat
-      [ "Expected status was ", show number
-      , " but received status was ", show $ H.statusCode status
-      , " for " <> (T.unpack $ formatRequestBuilderDataForDebugging latestRequest)
-      , if isUTF8ContentType
-          then ". For debugging, the body was: " <> (T.unpack $ YT.Internal.getBodyTextPreview body)
-          else ""
+    let mResponseContentType = lookup hContentType headers
+        isResponseUTF8ContentType = maybe False YT.Internal.contentTypeHeaderIsUtf8 mResponseContentType
+
+        responsePreview = T.unpack $ YT.Internal.getBodyTextPreview body
+        previewFinal = if responsePreview == "" then "<empty response>" else 
+                if isResponseUTF8ContentType then T.unpack $ YT.Internal.getBodyTextPreview body
+                    else "<content-type not suitable for printing>"
+
+    liftIO $ flip HUnit.assertBool (H.statusCode status == number) $ unlines
+      [ "Expected status: " <> show number
+      , "But status was: " <> (show $ H.statusCode status)
+      , "Request: " <> (T.unpack $ formatRequestBuilderDataForDebugging latestRequest)
+      , "Request body: " <> (T.unpack $ getRequestBodyPreview latestRequest)
+      , "Response body: " <> previewFinal
       ]
 
+  where
+    getRequestBodyPreview :: RequestBuilderData site -> T.Text
+    getRequestBodyPreview RequestBuilderData{..} = 
+        let mRequestContentType = lookup hContentType rbdHeaders
+            isRequestUTF8ContentType = maybe False YT.Internal.contentTypeHeaderIsUtf8 mRequestContentType
+        in case rbdPostData of
+                    MultipleItemsPostData xs -> case xs of
+                        [] -> "<empty body>"
+                        _ -> "<POST param preview not supported>"
```

Comment:
Ideal thing would be to support this, might work up the energy to extract all the multipart functions from `request`